### ISSUE TITLE
BUG: `.explore()` fails when `column` is passed a non-numerical list

### ISSUE THREAD

**Issue by olsgaard** [geopandas/geopandas#2587]
- [ x] I have checked that this issue has not already been reported.

- [x ] I have confirmed this bug exists on the latest version of geopandas.

- [x ] (optional) I have confirmed this bug exists on the main branch of geopandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

```python
import numpy as np
import geopandas as gpd
xs, ys =np.random.rand(20)+np.linspace(0,50, 20), np.random.rand(20)+np.linspace(0,50, 20)

gdf = gpd.GeoDataFrame({
    "cat": ["a", "b"]*10,
    "num" : range(20),
    "geometry": gpd.points_from_xy(xs, ys)
    }, crs="WGS84"
)

# works fine
gdf.explore(column="cat")
gdf.explore(column=gdf.num)

#Fails with a UFuncTypeError in numpy
gdf.explore(column=gdf.cat)
```

```
File ~/miniconda3/envs/gds/lib/python3.9/site-packages/numpy/core/function_base.py:127, in linspace(start, stop, num, endpoint, retstep, dtype, axis)
    123 div = (num - 1) if endpoint else num
    125 # Convert float/complex array scalars to float, gh-3504
    126 # and make sure one can use variables that have an __array_interface__, gh-6634
--> 127 start = asanyarray(start) * 1.0
    128 stop  = asanyarray(stop)  * 1.0
    130 dt = result_type(start, stop, float(num))

UFuncTypeError: ufunc 'multiply' did not contain a loop with signature matching types (dtype('<U1'), dtype('float64')) -> None
```

#### Problem description

Passing value to `column` as a reference to a column in gdf or as a pd.Series should yield the same result if the values are the same.

#### Expected Output

```
gdf.explore(column=gdf.cat)
```
Should output exactly the same as
```
gdf.explore(column="cat")
```

but former raises a `UFuncTypeError`

#### How to fix

Change [line 374 in `explore.py`](https://github.com/geopandas/geopandas/blob/main/geopandas/explore.py#L374) from `elif` to `if`.

What happens is, if `column` is list_like, all the checks for if column is categorical are skipped. This causes a series of object types to be multiplied by a float, and numpy has no way of doing that and raises a `UFuncTypeError`

**Comment by olsgaard** [comment]
I have a fix in a fork. Should I just start a PR? "It works on my machine", but I am unsure how to run all tests against CI

**Comment by rraymondgh** [comment]
@olsgaard I'm working on https://github.com/geopandas/geopandas/pull/2590 to improve this type of behaviour.  Have identified this issue,  haven't fixed yet...  Post into a comment fix that you have identified rather than two of us generating PRs that fix same thing

**Comment by rraymondgh** [comment]
@olsgaard did you find `elif pd.api.types.is_categorical_dtype(gdf[column]):` should be **if** ?  changing that tests still pass and passing a series now works

**Comment by olsgaard** [comment]
@rraymondgh , Yes, `elif pd.api.types.is_categorical_dtype(gdf[column]):` should be `if pd.api.types.is_categorical_dtype(gdf[column]):`
